modules/library/metadata/src/main/java/org/geotools/factory/GeoTools.java

```diff
@@ -786,7 +791,69 @@ public static Hints addDefaultHints(final Hints hints) {
         }
         return completed;
     }
-    
+
+    /**
+     * Returns the default entity resolver, used to configure {@link SAXParser}.
+     * 
+     * @param hints An optional set of hints, or {@code null} if none, see {@link Hints#ENTITY_RESOLVER}.
+     * @return An entity resolver (never {@code null})
+     */
+    public static EntityResolver getEntityResolver(Hints hints) {
```

Comment:
This method is too complex and does too many different things - break into multiple methods.